Gcp/service/memorystore

guandongzhang opened the pull request:
Sorry teacher, I previously forked a repository and submitted code from there. I have already cloned the repository and named it my own service name. Now I am resubmitting a new version of the PR, which is consistent with my original code.

Commit:
Google Secure Source Manager resource creation and relevant kms configuration check

Checking if the kms_key value is present in the resource creation plan or not. Key Management Service is used to encrypt resources or data. This can be managed by the customer instead of relying on the service provider

Commit:
Secure Source Manager instance creation with private configuration setting

private_config block checks if the given instance is private. By default, all instances are private as per the company policy.

Pat-Stuart committed:
Upgrade of helpers and new templates (#7)

+ Removal of AzureRM version to default to the latest version.

Commit:
Terraform and policy files for ca_pool attribute

Terraform and policy files for ca_pool attribute -

1. Includes compliant and non compliant terraform files
2. vars and policy rego files

Commit:
Updating policy file for kms_key attribute

Updating policy.rego for kms_key

Commit:
Updating policy file for private_config attribute

Updating policy.rego file for private_config attribute

Commit:
Terraform and policy files for workforce_identity_federation attribute

Files for workforce_identity_federation attribute -

1. Includes compliant and non compliant terraform files
2. vars and policy rego files

Commit:
Terraform and policy files for IAM binding attribute

Terraform and policy files for binding attribute -

1. Includes compliant and non compliant terraform files
2. vars and policy rego files

Commit:
Terraform and policy files for branch_rule attribute

Terraform and policy files for branch_rule attribute -

1. Includes compliant and non compliant terraform files
2. vars and policy rego files

Commit:
Terraform and policy files for require_comments_resolved attribute

Terraform and policy files for require_comments_resolved attribute -

1. Includes compliant and non compliant terraform files
2. vars and policy rego files

Commit:
Terraform and policy files for linear_history attribute

Terraform and policy files for linear_history attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for require_pull_request attribute

Terraform and policy files for require_pull_request attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for reviews_count attribute

Terraform and policy files for reviews_count attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for default_branch attribute

Terraform and policy files for default_branch attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for description attribute

Terraform and policy files for description attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for git ignore attribute

Terraform and policy files for  git ignore attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for license attribute

Terraform and policy files for  license attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for readme attribute

Terraform and policy files for  readme attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Adding read me file and the images used

Read me file to explain about the service and the resources available

Commit:
Terraform and policy files for policy_data attribute

Terraform and policy files for  policy_data attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Edited Read me file

JBarazani committed:
Added Item to git ignore

JBarazani committed:
Added to .gitignore

sthenusan committed:
Gcp/service/scc (#47)

* Upgrade of helpers and new templates (#7)

+ Removal of AzureRM version to default to the latest version.

* add Init changes for scc-google_scc_event_threat_detection_custom_module-enablement_state

* add git ignore

* Add rough structures for other policies

* add changes in display_name

* add changes in display_name

* Add correct files for event-enablement_state

* add files for folder_custom_module

* Add files for organization custom module policy

* Add files for project custom  module policy

* Add changes for best practice in coding

* Add files for length check for display_name in source

* Add fixes for feedback

* Add changes for feedback of code review

---------

Co-authored-by: Pat <71543754+Pat-Megabite@users.noreply.github.com>

sthenusan committed:
Revert "Gcp/service/scc (#47)" (#48)

This reverts commit 1ad5709047e59b62ae3b6cb756e41285015bb2b0.

Commit:
Fixing issues raised in the pull request.

1. Renamed the folders to match with the resources names
2. Removed the images folder and readme file
3. Fixed the indentation by running opa fmt
4. Removed policies that are not aligned with security (readme, description, default branch rule)
5. Fixed a policy to use pattern whitelist

Commit:
Merge branch 'gcp/secure-source-manager' of https://github.com/Hardhat-Enterprises/Policy-Deployment-Engine into gcp/secure-source-manager

Commit:
Removing plan.json files

Parsing JSON object for a whitelisting policy

sthenusan committed:
Gcp/service/scc (#11) (#18)

* Gcp/service/scc (#11)

* Add changes for policies of scc

* Add changes to remove display_name policiy as it dont work as expected

* Add changes for feebback

* Add changes to fix indendation issue

sumedh004 committed:
Gcp/service/parameter manager (#19)

* initial policies

* policies for all resources

* remove project from config.tf

* remove comments

* remove project from config

* folder cleanup

sumedh004 committed:
Gcp/service/cloud vmware engine (#22)

* initial policy for dlp

* temporary helpers.rego fix

* polciies for network resource

* polciies for network resource

* policies for network peering resource

* policies for network policy resource

* policies for external access rule

* merged helper rego

* policy rewrite for new helper

* fix policy

* policy for external access rule

* remove project from config

* restructure and minor changes

* folder cleanup

jashan-lefty committed:
changes for PR

guandongzhang committed:
init

guandongzhang committed:
init

guandongzhang committed:
init

guandongzhang committed:
main & memorystore policy

guandongzhang committed:
main & memorystore policy

guandongzhang committed:
main & memorystore policy

guandongzhang committed:
main & memorystore policy

guandongzhang committed:
Merge remote-tracking branch 'origin/gcp/service/Memorystore' into gcp/service/Memorystore

visal-dam committed:
Updated helpers.rego (#64)

* Updated helpers.rego

- Updated array_contains() to also check arrays
- Updated how resources can be identified (either just 'name' or within 'values' block)

All are backwards compatible.

* Update helpers.rego

(1) updated array_contains() to check for array inputs
-> whitelist: subsets
e.g.: 
1) in: [a, b, c], compliant: [a, b, c, d] -> true
2) in: [a, b, e] -> false
-> blacklist: intersections
e.g.g:
1) in: [a, b, d], non-compliant: [d, e, f] -> true
2) in: [a, b, c] -> false

(2) updated how resources can be identified using default "name" if none exists in "values".

* Update helpers.rego

commented out debug print statements

guandongzhang committed:
Merge remote-tracking branch 'origin/gcp/service/Memorystore' into gcp/service/Memorystore

guandongzhang committed:
Merge remote-tracking branch 'origin/gcp/service/Memorystore' into gcp/service/Memorystore

guandongzhang committed:
Copy my fork responsity into this branch

guandongzhang committed:
Copy fork responsity into gcp/service/Firebaserules

Include 5 policy.rego, 5 c.tf and 5 nc.tf

guandongzhang committed:
chore: remove all .DS_Store files

guandongzhang committed:
Merge branch 'dev' into gcp/service/Memorystore

# Conflicts:
#	README.md
#	inputs/gcp/cloud_vmware_engine/external_access_rule/block_broad_external_access/plan
#	inputs/gcp/cloud_vmware_engine/external_access_rule/block_broad_external_access/plan.json
#	inputs/gcp/cloud_vmware_engine/network/allowed_location/plan
#	inputs/gcp/cloud_vmware_engine/network/allowed_location/plan.json
#	inputs/gcp/cloud_vmware_engine/network/block_legacy/plan
#	inputs/gcp/cloud_vmware_engine/network/block_legacy/plan.json
#	inputs/gcp/cloud_vmware_engine/network_peering/block_custom_routes/plan
#	inputs/gcp/cloud_vmware_engine/network_peering/block_custom_routes/plan.json
#	inputs/gcp/cloud_vmware_engine/network_peering/block_thirdparty/plan
#	inputs/gcp/cloud_vmware_engine/network_peering/block_thirdparty/plan.json
#	inputs/gcp/cloud_vmware_engine/network_policy/allowed_location/plan
#	inputs/gcp/cloud_vmware_engine/network_policy/allowed_location/plan.json
#	inputs/gcp/cloud_vmware_engine/network_policy/block_external_ip/plan
#	inputs/gcp/cloud_vmware_engine/network_policy/block_external_ip/plan.json
#	inputs/gcp/cloud_vmware_engine/network_policy/block_internet_access/plan
#	inputs/gcp/cloud_vmware_engine/network_policy/block_internet_access/plan.json
#	inputs/gcp/cloud_vmware_engine/private_cloud/allowed_location/plan
#	inputs/gcp/cloud_vmware_engine/private_cloud/allowed_location/plan.json
#	inputs/gcp/cloud_vmware_engine/private_cloud/zonal_location/plan
#	inputs/gcp/cloud_vmware_engine/private_cloud/zonal_location/plan.json
#	inputs/gcp/parameter_manager/parameter/encryption/plan
#	inputs/gcp/parameter_manager/parameter/encryption/plan.json
#	inputs/gcp/parameter_manager/regional_parameter/allowed_location/plan
#	inputs/gcp/parameter_manager/regional_parameter/allowed_location/plan.json
#	inputs/gcp/parameter_manager/regional_parameter/encryption/plan
#	inputs/gcp/parameter_manager/regional_parameter/encryption/plan.json
#	inputs/gcp/scc/event_threat_detection_custom_module/enablement_state/.terraform.lock.hcl
#	inputs/gcp/scc/event_threat_detection_custom_module/enablement_state/plan
#	inputs/gcp/scc/event_threat_detection_custom_module/enablement_state/plan.json
#	inputs/gcp/scc/folder_custom_module/enablement_state/.terraform.lock.hcl
#	inputs/gcp/scc/folder_custom_module/enablement_state/plan
#	inputs/gcp/scc/folder_custom_module/enablement_state/plan.json
#	inputs/gcp/scc/mute_config/type/.terraform.lock.hcl
#	inputs/gcp/scc/mute_config/type/plan
#	inputs/gcp/scc/mute_config/type/plan.json
#	inputs/gcp/scc/organization_custom_module/enablement_state/.terraform.lock.hcl
#	inputs/gcp/scc/organization_custom_module/enablement_state/plan
#	inputs/gcp/scc/organization_custom_module/enablement_state/plan.json
#	inputs/gcp/scc/project_custom_module/enablement_state/.terraform.lock.hcl
#	inputs/gcp/scc/project_custom_module/enablement_state/plan
#	inputs/gcp/scc/project_custom_module/enablement_state/plan.json

guandongzhang committed:
fix some need

guandongzhang committed:
Merge branch 'dev' into gcp/service/Memorystore

# Conflicts:
#	inputs/gcp/cloud_vmware_engine/external_access_rule/block_broad_external_access/c.tf
#	inputs/gcp/cloud_vmware_engine/external_access_rule/block_broad_external_access/nc.tf
#	inputs/gcp/cloud_vmware_engine/network/allowed_location/c.tf
#	inputs/gcp/cloud_vmware_engine/network/allowed_location/nc.tf
#	inputs/gcp/cloud_vmware_engine/network/block_legacy/c.tf
#	inputs/gcp/cloud_vmware_engine/network/block_legacy/nc.tf
#	inputs/gcp/cloud_vmware_engine/network_peering/block_custom_routes/c.tf
#	inputs/gcp/cloud_vmware_engine/network_peering/block_custom_routes/nc.tf
#	inputs/gcp/cloud_vmware_engine/network_peering/block_thirdparty/c.tf
#	inputs/gcp/cloud_vmware_engine/network_peering/block_thirdparty/nc.tf
#	inputs/gcp/cloud_vmware_engine/network_policy/allowed_location/c.tf
#	inputs/gcp/cloud_vmware_engine/network_policy/allowed_location/nc.tf
#	inputs/gcp/cloud_vmware_engine/network_policy/block_external_ip/c.tf
#	inputs/gcp/cloud_vmware_engine/network_policy/block_external_ip/nc.tf
#	inputs/gcp/cloud_vmware_engine/network_policy/block_internet_access/c.tf
#	inputs/gcp/cloud_vmware_engine/network_policy/block_internet_access/nc.tf
#	inputs/gcp/cloud_vmware_engine/private_cloud/allowed_location/c.tf
#	inputs/gcp/cloud_vmware_engine/private_cloud/allowed_location/nc.tf
#	inputs/gcp/cloud_vmware_engine/private_cloud/zonal_location/c.tf
#	inputs/gcp/cloud_vmware_engine/private_cloud/zonal_location/nc.tf
#	inputs/gcp/google_kms/google_kms_crypto_key/scheduled_destroy_duration/config.tf
#	inputs/gcp/parameter_manager/parameter/encryption/c.tf
#	inputs/gcp/parameter_manager/parameter/encryption/nc.tf
#	inputs/gcp/parameter_manager/regional_parameter/allowed_location/c.tf
#	inputs/gcp/parameter_manager/regional_parameter/allowed_location/nc.tf
#	inputs/gcp/parameter_manager/regional_parameter/encryption/c.tf
#	inputs/gcp/parameter_manager/regional_parameter/encryption/nc.tf
#	inputs/gcp/scc/event_threat_detection_custom_module/enablement_state/c.tf
#	inputs/gcp/scc/event_threat_detection_custom_module/enablement_state/nc.tf
#	inputs/gcp/scc/folder_custom_module/enablement_state/c.tf
#	inputs/gcp/scc/folder_custom_module/enablement_state/nc.tf
#	inputs/gcp/scc/organization_custom_module/enablement_state/c.tf
#	inputs/gcp/scc/organization_custom_module/enablement_state/nc.tf
#	inputs/gcp/scc/project_custom_module/enablement_state/c.tf
#	inputs/gcp/scc/project_custom_module/enablement_state/nc.tf

guandongzhang committed:
fix some bug

guandongzhang committed:
fix bug:Terraform failed to compile!

guandongzhang committed:
fix bug:Terraform failed to compile![2.0]

guandongzhang committed:
fix bug:Terraform failed to compile![2.0]

guandongzhang committed:
fix bug:Terraform failed to compile![3.0]

guandongzhang committed:
fix bug:Terraform failed to compile![4.0]

guandongzhang committed:
fix bug:Terraform failed to compile![5.0]

guandongzhang committed:
fix bug:Terraform failed to compile![6.0]

guandongzhang committed:
Merge branch 'dev' into gcp/service/Memorystore

# Conflicts:
#	.gitignore
#	policies/gcp/cloud_vmware_engine/external_access_rule/block_broad_external_access/policy.rego
#	policies/gcp/cloud_vmware_engine/external_access_rule/vars.rego
#	policies/gcp/cloud_vmware_engine/network/allowed_location/policy.rego
#	policies/gcp/cloud_vmware_engine/network/block_legacy/policy.rego
#	policies/gcp/cloud_vmware_engine/network/vars.rego
#	policies/gcp/cloud_vmware_engine/network_peering/block_custom_routes/policy.rego
#	policies/gcp/cloud_vmware_engine/network_peering/block_thirdparty/policy.rego
#	policies/gcp/cloud_vmware_engine/network_peering/vars.rego
#	policies/gcp/cloud_vmware_engine/network_policy/allowed_location/policy.rego
#	policies/gcp/cloud_vmware_engine/network_policy/block_external_ip/policy.rego
#	policies/gcp/cloud_vmware_engine/network_policy/block_internet_access/policy.rego
#	policies/gcp/cloud_vmware_engine/network_policy/vars.rego
#	policies/gcp/cloud_vmware_engine/private_cloud/allowed_location/policy.rego
#	policies/gcp/cloud_vmware_engine/private_cloud/vars.rego
#	policies/gcp/cloud_vmware_engine/private_cloud/zonal_location/policy.rego
#	policies/gcp/parameter_manager/parameter/encryption/policy.rego
#	policies/gcp/parameter_manager/parameter/vars.rego
#	policies/gcp/parameter_manager/regional_parameter/allowed_location/policy.rego
#	policies/gcp/parameter_manager/regional_parameter/encryption/policy.rego
#	policies/gcp/parameter_manager/regional_parameter/vars.rego
#	policies/gcp/scc/event_threat_detection_custom_module/enablement_state/policy.rego
#	policies/gcp/scc/event_threat_detection_custom_module/vars.rego
#	policies/gcp/scc/folder_custom_module/enablement_state/policy.rego
#	policies/gcp/scc/folder_custom_module/vars.rego
#	policies/gcp/scc/organization_custom_module/enablement_state/policy.rego
#	policies/gcp/scc/organization_custom_module/vars.rego
#	policies/gcp/scc/project_custom_module/enablement_state/policy.rego
#	policies/gcp/scc/project_custom_module/vars.rego

guandongzhang committed:
fix bug: can't find format_range_validation_message in _helper.rego

guandongzhang committed:
fix bug: c & nc name

Pat-Stuart commented on policies/gcp/_helpers/helpers.rego:
Don't make changes to the helpers

guandongzhang replied:
I have reverted helpers and auto_test, but i found a bug in helpers: it lack a function and now all the policy can't pass the test.

Pat-Stuart replied:
I can still see changes, this bug has occurred before when changes have been made to the helpers previously. I can't see the changes in the PR and the commit history is too bizarre for me to investigate. I would suggest going to the file in the dev branch and just replacing your helpers with its contents to restore correct functionality.

guandongzhang replied:
Thanks, I use the latest helpers.rego in dev yesterday and found it has been fix the bug.

Pat-Stuart commented on policies/gcp/memorystore/memorystore_instance/deletion_protection_config/policy.rego:
```diff
@@ -0,0 +1,26 @@
+        },
+        {
+            "condition": "Checks if deletion_protection_config.enable_deletion_protection is true",
+            "attribute_path": ["deletion_protection_config", "enable_deletion_protection"],
```
This attribute path is incorrect it is just enable_deletion_protection

guandongzhang replied:
Thanks. Modified.

Pat-Stuart commented on policies/gcp/memorystore/memorystore_instance/persistence_config/policy.rego:
This policy while I agree it is security related, can't currently be accessed via our helpers due to being a object in an array. You can make a note of this in your documentation but please remove it.

guandongzhang replied:
Thanks. I will make a note in my doc.

Pat-Stuart reviewed (changes requested):
All of the .idea folder should be deleted. You can remove the empty folders in your PR.

The changes to the helpers, templates, auto_test and other people work needs to be reverted. You can do this through a git reset soft and force push or manually you can look this up.

You have massively over-complicated your TF examples by copy and pasting the advanced example on the registry, there are several things in there causing your TF to not compile. I have attached an example of what we would expect to see for your deletion protection policy. This keeps it simple for you to understand and us to review. This works for the policy and compiles.

<img width="1316" height="1052" alt="image" src="https://github.com/user-attachments/assets/95690ba0-3d3f-41d1-8a51-f9064cab25df" />
Figure 1: c.tf

<img width="744" height="366" alt="image" src="https://github.com/user-attachments/assets/e216d83f-5b1c-4c88-a73f-4d92bd98d52e" />
Figure 2: config.tf, unchanged from template

<img width="665" height="382" alt="image" src="https://github.com/user-attachments/assets/3963fe7f-76d1-48a7-b15a-64ac3757e66d" />
Figure 3: nc.tf

guandongzhang committed:
Merge branch 'dev' into gcp/service/Memorystore

guandongzhang committed:
reverted change to the helpers, auto_test

guandongzhang committed:
1.simplify c.tf & nc.tf
2.fix bug in delection_proction_config: attribute path incorrect
3.remove persistence_config/policy.rego, and add a note in doc

guandongzhang committed:
remove persistence_config:config.tf

guandongzhang committed:
remove persistence_config:config.tf

guandongzhang committed:
For test

guandongzhang committed:
reverted change

guandongzhang committed:
Merge branch 'dev' into gcp/service/Memorystore

# Conflicts:
#	.DS_Store
#	inputs/.DS_Store
#	inputs/gcp/.DS_Store
#	inputs/gcp/scc/.DS_Store
#	policies/.DS_Store
#	policies/gcp/.DS_Store
#	policies/gcp/scc/.DS_Store

guandongzhang committed:
delete useless code

guandongzhang committed:
fix some bugs

guandongzhang committed:
revert to vision: before merge dev

Pat-Stuart commented:
There is still files that need to be reverted notably the config.tf files for other services

guandongzhang committed:
Copy the raw file of helper.rego in dev.

guandongzhang commented:
> There is still files that need to be reverted notably the config.tf files for other services

Thanks. I will merge the dev in my branch again.

Pat-Stuart commented:
You will need to do a reset or add the files back manually, a merge won't work as your change is downstream in git

guandongzhang committed:
Merge branch 'dev' into gcp/service/Memorystore

guandongzhang committed:
add the files back

guandongzhang committed:
fix bugs:
Service: memorystore | Resource: memorystore_instance | Policy: deletion_protection_config
Terraform failed to compile!

Service: memorystore | Resource: memorystore_instance | Policy: authorization_mode
Terraform failed to compile!

guandongzhang committed:
fix bugs:
Service: memorystore | Resource: memorystore_instance | Policy: deletion_protection_config
Terraform failed to compile!

guandongzhang committed:
fix bugs:
Service: memorystore | Resource: memorystore_instance | Policy: deletion_protection_config
Terraform failed to compile!

Service: memorystore | Resource: memorystore_instance | Policy: authorization_mode
Terraform failed to compile!

guandongzhang committed:
fix bugs:
Service: memorystore | Resource: memorystore_instance | Policy: deletion_protection_config
Terraform failed to compile!

Service: memorystore | Resource: memorystore_instance | Policy: authorization_mode
Terraform failed to compile!

guandongzhang committed:
fix bugs:
Service: memorystore | Resource: memorystore_instance | Policy: deletion_protection_config
Terraform failed to compile!

Service: memorystore | Resource: memorystore_instance | Policy: authorization_mode
Terraform failed to compile!

guandongzhang committed:
fix bugs:
Service: memorystore | Resource: memorystore_instance | Policy: deletion_protection_config
Terraform failed to compile!

Service: memorystore | Resource: memorystore_instance | Policy: authorization_mode
Terraform failed to compile!

guandongzhang committed:
fix bugs:
Service: memorystore | Resource: memorystore_instance | Policy: deletion_protection_config
Terraform failed to compile!

Service: memorystore | Resource: memorystore_instance | Policy: authorization_mode
Terraform failed to compile!

guandongzhang commented:
@Pat-Stuart 
Hi teacher, all my policy can be passed in my local env, but it failed in CI, fail log:
"Service: memorystore | Resource: memorystore_instance | Policy: deletion_protection_config
Terraform failed to compile!

Service: memorystore | Resource: memorystore_instance | Policy: authorization_mode
Terraform failed to compile!"

I see the auto_test file, it said the failure may cause by [terraform init & terraform plan -out=tfplan] failed. Reason may be provider version ≤ 6.15.

I want to ask:
1.where can I see the provider version in CI env?
2.I am sure my code is right, but i can't pass the test now, will my code merge in dev?

My test pictures in local env are as followed:
<img width="1080" height="689" alt="testcase1" src="https://github.com/user-attachments/assets/74e1b92f-6382-4c3b-8b69-1b647c623167" />
<img width="1080" height="689" alt="testcase2" src="https://github.com/user-attachments/assets/cc1f5600-f566-4b74-ba46-a2cd270eed4c" />

Pat-Stuart commented:
It pulls the latest TF version. 
Your policies are fine I just need you to restore the other files deleted check the files changed tab. I am only concerned about the config.tf for other services removed.

guandongzhang committed:
Merge branch 'dev' into gcp/service/Memorystore

# Conflicts:
#	inputs/gcp/google_dataproc_on_gdc/application_environment/spark_application_environment_config/main.tf

guandongzhang committed:
restore delete file
Use command：
git diff --name-status origin/dev | ForEach-Object {
    if ($_ -match "^D\s+(.+)$") {
        git checkout origin/dev -- $matches[1]
    }
}

guandongzhang commented:
> It pulls the latest TF version. Your policies are fine I just need you to restore the other files deleted check the files changed tab. I am only concerned about the config.tf for other services removed.

@Pat-Stuart 
Thank you. I have restored all the delete file compare my branch with dev branch.
Use command: 
git diff --name-status origin/dev | ForEach-Object {
    if ($_ -match "^D\s+(.+)$") {
        git checkout origin/dev -- $matches[1]
    }
}
Now no other service config.tf or other files has been deleted by my branch.

Pat-Stuart reviewed (dismissed):
LGTM
Thanks for your work
